proto/dataplane.proto

```diff
@@ -0,0 +1,214 @@
+syntax = "proto3";
+
+package config;
+
+option go_package = "dataplane/grpc";
+
+service ConfigService {
+  rpc GetConfig (GetConfigRequest) returns (GatewayConfig);
+  rpc GetConfigGeneration (GetConfigGenerationRequest) returns (GetConfigGenerationResponse);
+  rpc UpdateConfig (UpdateConfigRequest) returns (UpdateConfigResponse);
+}
+
+message GetConfigRequest {
+}
+
+message UpdateConfigRequest {
+  GatewayConfig config = 1;
+}
+
+message UpdateConfigResponse {
+  Error error = 1;
+  string message = 2;
+}
+
+enum Error {
+  ERROR_NONE = 0;
+  ERROR_VALIDATION_FAILED = 1;
+  ERROR_APPLY_FAILED = 2;
+  ERROR_UNKNOWN_ERROR = 3;
+}
+
+message GetConfigGenerationRequest {
+}
+
+message GetConfigGenerationResponse {
+  uint64 generation = 1;
+}
+
+/* ================================================ */
+/* ==== Interfaces (both underlay and overlay) ==== */
+/* ================================================ */
+
+message Interface {
+  string name = 1;
+  string ipaddr = 2;
+  IfType type = 3;
+  IfRole role = 4;
+  uint32 vlan = 5;
+  string macaddr = 6;
```

Comment:
How the MAC address should be constructed for the interfaces?

Reply:
This should be something similar to ESLAG MAC base. Will be used by dataplane to generate unified MAC address for VTEP and bridging